frontend/frontend-react/src/components/CreateRackForm.js

```diff
@@ -63,48 +28,76 @@ export class CreateRackForm extends Component {
   handleSubmit = (e) => {
     e.preventDefault();
 
-    let rack_num = this.state.rack_number;
+    let start_rack = this.state.rack_num_start;
+    let end_rack = this.state.rack_num_end;
 
+    // let singleFirstState = Object.assign({}, this.state.id);
+    // let singleStateCopy = Object.assign(singleFirstState, this.state.rackNumber);
     let stateCopy = Object.assign({}, this.state);
+    console.log(stateCopy);
+
     let stateToSend = this.removeEmpty(stateCopy);
 
+    // let multipleFirstState = Object.assign({}, this.state.rack_num_start)
+
     const validNumRegex = new RegExp("^[A-Z]\\d+$", 'i');
 
-    if ((validNumRegex.test(rack_num))) {
-      axios.post('/api/racks/', stateToSend)
-        .then(function (response) {
-          //   let message = response.data.results;
-          alert('Creation of ' + rack_num + ' was successful.');
-        })
-        .catch(function (error) {
-          alert('Creation was not successful.\n' + JSON.stringify(error.response.data, null, 2));
-        });
-    } else {
-      alert("Rack Numbers must be specified by a Single Letter Followed by Multiple Numbers.");
+    if(start_rack !== null && (end_rack == null || end_rack === '')){
+      if ((validNumRegex.test(start_rack))) {
+        axios.post('/api/racks/', stateToSend)
+          .then(function (response) {
+            //   let message = response.data.results;
+            alert('Creation of ' + start_rack + ' was successful.');
+          })
+          .catch(function (error) {
+            alert('Creation was not successful.\n' + JSON.stringify(error.response.data, null, 2));
+          });
+      } else {
+        // console.log(this.state.rack_number);
+        alert("Rack Numbers must be specified by a Single Letter Followed by Multiple Numbers.");
+      }
+    }
+    else if(start_rack !== null && end_rack !== null){
+      if ((validNumRegex.test(start_rack) && validNumRegex.test(end_rack))) {
+        axios.post('/api/racks/many/', stateToSend)
+          .then(function (response) {
+            let message = response.data.results;
+            alert(response.data.results);
+            this.props.sendShowTable(true);
+          })
+          .catch(function (error) {
+            alert('Creation was not successful.\n' + JSON.stringify(error.response.data, null, 2));
+          });
+      } else {
+        alert("Rack Numbers must be specified by a Single Letter Followed by Multiple Numbers.");
+      }
     }
-    this.props.sendShowTable(true);
   }
 
   render() {
     return (
-      <div>
-        <Button variant="outlined" onClick={() => this.props.sendShowTable(true)}>Back</Button>{' '}
-        <form onSubmit={this.handleSubmit}>
+      <Container maxwidth="s">
+        <Link to='/racks/'><Button variant="outlined">Back</Button>{' '}</Link>
+        <form onSubmit={this.handleSubmit} action='/racks/'>
```

Comment:
Do you think in addition to wrapping this in a container you could also put an empty grid row in just so the spacing isn't super tight on top? You can look at the other create/edit forms for inspiration

![image](https://user-images.githubusercontent.com/40015320/74575374-c6d64600-4f54-11ea-8c5a-255ae12b8f92.png)

Reply:
Could delete all the back buttons for now since we can route

Reply:
Yeah having a cancel button works just as well here